be/src/storage/rowset/column_reader.h

```diff
@@ -247,6 +255,21 @@ class ColumnReader {
 
     using SubReaderList = std::vector<std::unique_ptr<ColumnReader>>;
     std::unique_ptr<SubReaderList> _sub_readers;
+    // only used for struct column right now
+    struct SubReaderId {
+        std::string name;
+        int32_t id;
```

Comment:
why are both name and id needed? IMO id is enough to identify a specific field.

Reply:
> why are both name and id needed? IMO id is enough to identify a specific field.

For historical data, the unique ID of sub_column is always -1, but the name is unique. Therefore, we need to distinguish historical data by name. After supporting add/drop field operations for the struct column, the following scenarios might occur:

1. Drop field v1.
2. Add field v1.

The v1 in step 2 is different from the v1 in step 1 and needs to be distinguished. Therefore, we need to use the ID to make this distinction.

Another feasible solution is to always use the name for distinction. In the second step, a unique ID can be added as a suffix to the name of the newly added field v1 (e.g., v1_1). However, this means that the field name returned in the query results would be v1_1 instead of v1, which I believe could be confusing for users unless we convert the field name back during the query

Reply:
It is better to comment on why the name is needed.